asreview/webapp/utils/project.py

```diff
@@ -1,564 +0,0 @@
-
-            # extract all files to a temporary folder
-            tmpdir = tempfile.mkdtemp()
-            zip_obj.extractall(path=tmpdir)
-
-    except zipfile.BadZipFile:
-        raise ValueError("File is not an ASReview file.")
-
-    try:
-        # Open the project file and check the id. The id needs to be
-        # unique, otherwise it is exended with -copy.
-        import_project = None
-        fp = Path(tmpdir, "project.json")
-        with open(fp, "r+") as f:
-
-            # load the project info in scope of function
-            import_project = json.load(f)
-
-            # If the uploaded project already exists,
-            # then overwrite project.json with a copy suffix.
-            while is_project(import_project["id"]):
-                # project update
-                import_project["id"] = f"{import_project['id']}-copy"
-                import_project["name"] = f"{import_project['name']} copy"
-            else:
-                # write to file
-                f.seek(0)
-                json.dump(import_project, f)
-                f.truncate()
-
-        # location to copy file to
-        fp_copy = get_project_path(import_project["id"])
-        # Move the project from the temp folder to the projects folder.
-        os.replace(tmpdir, fp_copy)
-
-    except Exception:
-        # Unknown error.
-        raise ValueError("Failed to import project "
-                         f"'{file_name.filename}'.")
-
-    project_info = {}
-    project_info["id"] = import_project["id"]
-    project_info["name"] = import_project["name"]
-
-    return project_info
-
-
-def add_dataset_to_project(project_id, file_name):
-    """Add file path to the project file.
-
-    Add file to data subfolder and fill the pool of iteration 0.
-    """
-    project_path = get_project_path(project_id)
-    project_info = get_project_config(project_id)
-
-    update_project_info(project_id, dataset_path=file_name)
-
-    # fill the pool of the first iteration
-    as_data = read_data(project_id)
-
-    state_file = get_state_path(project_path)
-
-    with open_state(state_file, read_only=False) as state:
-
-        # save the record ids in the state file
-        state.add_record_table(as_data.record_ids)
-
-        # if the data contains labels, add them to the state file
-        # except for simulation project
-        if as_data.labels is not None and project_info["mode"] != PROJECT_MODE_SIMULATE:
-
-            labeled_indices = np.where(as_data.labels != LABEL_NA)[0]
-            labels = as_data.labels[labeled_indices].tolist()
-            labeled_record_ids = as_data.record_ids[labeled_indices].tolist()
-
-            # add the labels as prior data
-            state.add_labeling_data(
-                record_ids=labeled_record_ids,
-                labels=labels,
-                notes=[None for _ in labeled_record_ids],
-                prior=True
-            )
```

Comment:
@J535D165 This was not completely ported to the new `project.py`. This condition `project_info["mode"] != PROJECT_MODE_SIMULATE` is missing.